compiler/src/dotty/tools/dotc/config/Settings.scala

```diff
@@ -54,18 +59,21 @@ object Settings:
   }
 
   case class Setting[T: ClassTag] private[Settings] (
+    category: String,
     name: String,
     description: String,
     default: T,
     helpArg: String = "",
     choices: Option[Seq[?]] = None,
-    prefix: String = "",
+    prefix: Option[String] = None,
```

Comment:
What's the difference between Some("") and None?

Reply:
If the prefix of a setting exists, then this setting behaves differently; it is a different type of setting. API users can choose not to provide this prefix, thus not using this feature.  

Sole `String` does not express the property of being optional. String would mean that it either has an empty prefix (so that would include all possible settings or some other feature) or that there is no prefix. Option type makes it clear that the prefix can be not present. It is similar to the conflict of `None` vs`Some(null)`.

Reply:
I can't follow you.  Can you give 3 reasonable examples, with None, Some(""), and Some("-Whatever")?  Might be easier to follow your reasoning.

Reply:
Generally in the compiler we prefer sentinels over `Option`. That's the adopted code style. `Option` can be useful sometimes, but as Dale says we should not reach for it blindly.